model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/InfinispanUserSessionProvider.java

```diff
@@ -208,6 +209,25 @@ public void migrate(String modelVersion) {
         if ("26.0.0".equals(modelVersion)) {
             log.debug("Clear caches to migrate to Infinispan Protostream");
             CompletionStages.join(session.getProvider(InfinispanConnectionProvider.class).migrateToProtoStream());
+        } else if ("26.4.0".equals(modelVersion)) {
+            log.debug("Clear caches as client session entries are now outdated and are not migrated");
+            // This is a best-effort approach: Even if due to a rolling update some entries are left there, the checking of sessions and tokens does not depend on them.
+            // Refreshing of tokens will still work even if the user session does not contain the list of client sessions.
+            // This still keeps the user session cache to keep users logged in on a best effort basis. Only the offline user sessions cache is cleared.
```

Comment:
Is "Only the offline user sessions cache is cleared." outdated, or am I missing something?

Reply:
It is "Only the offline user sessions cache is cleared, but not the regular user sessions cache is cleared. All client session caches regular and offline client sessions are cleared as usual" ... yes, that's even longer.